### PR TITLE
fix(ds):  if value is dateTime, convert it to ISOString 

### PR DESCRIPTION
**Pull request by goombeer**
# Background

https://tailorpartnershq.slack.com/archives/C0581HGGLH3/p1715317032778909?thread_ts=1715309869.820629&cid=C0581HGGLH3

# Changes
[packages/design-systems/package.json](https://github.com/tailor-platform/frontend-packages/pull/200/files#diff-b3c3ba59c3ba445b0f4075c962272f6058d02fa85b06b8732963cba1b4aa7cf9R78): Using dayjs if value is valid.

[packages/design-systems/src/components/composite/Datagrid/SearchFilter/CustomFilter.tsx](https://github.com/tailor-platform/frontend-packages/pull/200/files#diff-ba24f0046bbb65e4b433f2c85c6fc8864b22af8057ca5adada1149daaab8e17cR379):If the value is dateTime, it has been modified to convert to ISOString.Also, refactored so that switch statements can be written in MetaType, since there seemed to be more if statements.

# Question
[Here's a test](https://github.com/tailor-platform/frontend-packages/blob/bbfedaf5d2782ffdf1450f5f96ada0a2cdf435cf/packages/design-systems/src/components/composite/Datagrid/SearchFilter/CustomFilter.test.tsx#L480), it looks like the number 800 is correct instead of the string "800", is it correct to return the string as a specification?

### CODE REVIEW

File: packages/design-systems/src/components/composite/Datagrid/SearchFilter/CustomFilter.tsx [hunk 1]
```diff
@@ -376,40 +377,83 @@ export const CustomFilter = <TData extends Record<string, unknown>>(
...
+        }
+        switch (metaType) {
+          case "boolean":
+            graphQLQueryObject[key] = generateGraphQLQueryObject(
+              isExitJointCondition,
+              value.toLowerCase() === "true",
+            );
+            break;
+          case "dateTime": {
+            const date = dayjs(value);
```
Comment (goombeer) [inline packages/design-systems/src/components/composite/Datagrid/SearchFilter/CustomFilter.tsx:422]: I used dayjs to check if value is a valid value.
In the following example, it does not behave properly,..,
```
const date1 = Date.parse('2024-02-30 23:00');
const date2 = new Date('2024-02-30 23:00');

console.log(date1); > 1709301600000
console.log(date2); > Fri Mar 01 2024 23:00:00 GMT+0900 (日本標準時)


```

File: packages/design-systems/src/components/composite/Datagrid/SearchFilter/CustomFilter.tsx [hunk 2]
```diff
@@ -424,18 +468,22 @@ export const CustomFilter = <TData extends Record<string, unknown>>(
     filterRows.forEach((row) => {
       if (row.currentState) {
         const { column, condition, value } = row.currentState;
+        const metaType = columns.find((c) => c.accessorKey === column)?.meta
+          ?.type as MetaType | undefined;
```
Comment (goombeer) [inline packages/design-systems/src/components/composite/Datagrid/SearchFilter/CustomFilter.tsx:472]: I would really like to modify the following type definitions, but the scope of influence was a bit large, so this is what I did.

https://github.com/tailor-platform/frontend-packages/blob/db7109141133718bbd61b4fb1a2f7c3c910c51a6/packages/design-systems/src/components/composite/Datagrid/type.d.ts#L25

File: packages/design-systems/src/components/composite/Datagrid/SearchFilter/CustomFilter.tsx [hunk 3]
```diff
@@ -424,18 +468,22 @@ export const CustomFilter = <TData extends Record<string, unknown>>(
...
           addToGraphQLQueryFilterRecursively(
             row.currentState,
             newFilterRowsState,
+            metaType,
           );
         }
       }
     });
     setFilterRowsState(newFilterRowsState);
+    // eslint-disable-next-line react-hooks/exhaustive-deps
```
Comment (goombeer) [inline packages/design-systems/src/components/composite/Datagrid/SearchFilter/CustomFilter.tsx:486]: If columns add deps, infinite loop occurs.

File: packages/design-systems/test-setup.js [hunk 4]
```diff
@@ -0,0 +1,13 @@
+const initialTimezone = process.env.TZ;
+let teardownHappened = false;
+
+export const setup = async () => {
```
Comment (goombeer) [inline packages/design-systems/test-setup.js:4]: The time zone in CI was UTC, so I changed the time zone.

File: packages/design-systems/src/components/composite/Datagrid/SearchFilter/CustomFilter.tsx (hunk 2 above)
Comment (IzumiSy) [reply packages/design-systems/src/components/composite/Datagrid/SearchFilter/CustomFilter.tsx:477]: Can you update the `type` to be `MetaType` as well in this PR. I have wanted to do so, but now is the timing. IMA will be okay on that.

File: packages/design-systems/package.json [hunk 5]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "@tailor-platform/design-systems",
-  "version": "0.28.0",
+  "version": "0.28.1",
```
Comment (IzumiSy) [inline packages/design-systems/package.json:3]: This PR now includes update on types (`MetaType`), so can you update here as the minor update?

```suggestion
  "version": "0.29.0",
```

**Review by IzumiSy — CHANGES_REQUESTED** [review]

File: packages/design-systems/src/components/composite/Datagrid/SearchFilter/CustomFilter.tsx (hunk 2 above)
Comment (goombeer) [reply packages/design-systems/src/components/composite/Datagrid/SearchFilter/CustomFilter.tsx]: Update type to  MetaType!
I was able to correct the problem successfully, and the correction area is now narrower😇
[fix(ds): update type to MetaType](https://github.com/tailor-platform/frontend-packages/pull/200/commits/5afce63387eed315922c66140c04d9f5681f557c)

**Pull request merged**